perllib/FixMyStreet/App/Form/Waste/Bulky/Amend.pm

```diff
@@ -0,0 +1,295 @@
+has_page choose_date_earlier => (
+    fields => [
+        'continue',         'chosen_date',
+        'show_later_dates',
+        ],
+    title => 'Choose date for collection',
+    template => 'waste/bulky/choose_date.html',
+    next => sub {
+        ( $_[1]->{later_dates} )
+            ? 'choose_date_later'
+            : 'add_items';
+    },
+);
+
+has_page choose_date_later => (
+    fields => [
+        'continue',         'chosen_date',
+        'show_earlier_dates',
+        ],
+    title => 'Choose date for collection',
+    template => 'waste/bulky/choose_date.html',
+    next => sub {
+        ( $_[1]->{earlier_dates} )
+            ? 'choose_date_earlier'
+            : 'add_items';
+    },
+);
+
+has_page location => (
+    title    => 'Location details',
+    fields   =>
+        [ 'location', 'location_photo', 'location_photo_fileid', 'continue' ],
+    next => 'summary',
+    update_field_list => sub {
+        my ($form) = @_;
+        my $fields = {};
+        $form->update_photo('location_photo', $fields);
+        return $fields;
+    },
+);
+
+has_page summary => (
+    fields => ['submit', 'tandc'],
+    title => 'Submit collection booking',
+    template => 'waste/bulky/summary.html',
+    next => sub { $_[0]->{no_slots} ? 'choose_date_earlier' : 'done' },
+    # Return to 'Choose date' page if slot has been taken in the meantime.
+    # Otherwise, proceed to payment.
+    pre_finished => sub {
+        my $form = shift;
+        my $c = $form->c;
+
+        ### DATE is Pboro specific
+        my $current_date = $c->stash->{amending_booking}->get_extra_field_value('DATE');
+        return 1 if $current_date eq $form->saved_data->{chosen_date};
+
+        my $slot_still_available
+            = $c->cobrand->call_hook(
+            check_bulky_slot_available => $form->saved_data->{chosen_date} );
+
+        return 1 if $slot_still_available;
+
+        # Clear date cache so user gets updated selection
+        my $uprn = $c->stash->{property}{uprn};
+        for (qw/earlier later/) {
+            delete $c->session->{"peterborough:bartec:available_bulky_slots:$_:$uprn"};
+        }
+        $c->stash->{flash_message} = 'choose_another_date';
+        $form->saved_data->{no_slots} = 1;
+        return 0;
+    },
+    finished => sub {
+        return $_[0]->wizard_finished('process_bulky_amend');
+    },
+
+);
+
+has_page done => (
+    title => 'Collection booked',
+    template => 'waste/bulky/confirmation.html',
+);
+
+
+has_field continue => (
+    type => 'Submit',
+    value => 'Continue',
+    element_attr => {
+        class      => 'govuk-button',
+        formaction => '?',
+    },
+    order => 999,
+);
+
+has_field submit => (
+    type => 'Submit',
+    value => 'Amend booking',
+    element_attr => { class => 'govuk-button' },
+    order => 999,
+);
+
+has_field chosen_date => (
+    type                 => 'Select',
+    widget               => 'RadioGroup',
+    label                => 'Available dates',
```

Comment:
This shows the booked date twice:

![image](https://github.com/mysociety/fixmystreet/assets/4776/56edafa2-9fac-4900-8d7f-ae368f7afb9a)